fatal error: olm/pk.h: No such file or directory during pip install

I'm trying to build this according to the [bridge setup](https://github.com/tulir/mautrix-signal/wiki/Bridge-setup) docs, but perhaps I've borked my environment? Here's what I see:

```
$ pip install --upgrade https://github.com/tulir/mautrix-signal/tarball/master#egg=mautrix-signal[all]
Collecting mautrix-signal[all]
  Using cached https://github.com/tulir/mautrix-signal/tarball/master
Collecting ruamel.yaml<0.17,>=0.15.35
  Using cached ruamel.yaml-0.16.12-py2.py3-none-any.whl (111 kB)
Collecting python-magic<0.5,>=0.4
  Using cached python_magic-0.4.18-py2.py3-none-any.whl (8.6 kB)
Collecting commonmark<0.10,>=0.8
  Using cached commonmark-0.9.1-py2.py3-none-any.whl (51 kB)
Collecting aiohttp<4,>=3
  Using cached aiohttp-3.7.2-cp37-cp37m-manylinux2014_x86_64.whl (1.3 MB)
Collecting yarl<2,>=1
  Downloading yarl-1.6.3-cp37-cp37m-manylinux2014_x86_64.whl (294 kB)
     |████████████████████████████████| 294 kB 1.6 MB/s 
Collecting attrs>=19.1
  Using cached attrs-20.3.0-py2.py3-none-any.whl (49 kB)
Collecting mautrix==0.8.0rc4
  Using cached mautrix-0.8.0rc4-py3-none-any.whl (219 kB)
Collecting asyncpg<0.22,>=0.20
  Using cached asyncpg-0.21.0-cp37-cp37m-manylinux1_x86_64.whl (2.7 MB)
Collecting python-olm<4,>=3
  Using cached python-olm-3.1.3.tar.gz (27 kB)
Collecting phonenumbers<9,>=8
  Using cached phonenumbers-8.12.13-py2.py3-none-any.whl (2.6 MB)
Collecting Pillow<8,>=4
  Using cached Pillow-7.2.0-cp37-cp37m-manylinux1_x86_64.whl (2.2 MB)
Collecting pycryptodome<4,>=3
  Using cached pycryptodome-3.9.9-cp37-cp37m-manylinux1_x86_64.whl (13.7 MB)
Collecting prometheus_client<0.9,>=0.6
  Using cached prometheus_client-0.8.0-py2.py3-none-any.whl (53 kB)
Collecting qrcode<7,>=6
  Using cached qrcode-6.1-py2.py3-none-any.whl (31 kB)
Collecting unpaddedbase64<2,>=1
  Using cached unpaddedbase64-1.1.0-py2.py3-none-any.whl (3.3 kB)
Collecting ruamel.yaml.clib>=0.1.2; platform_python_implementation == "CPython" and python_version < "3.9"
  Using cached ruamel.yaml.clib-0.2.2-cp37-cp37m-manylinux1_x86_64.whl (547 kB)
Collecting async-timeout<4.0,>=3.0
  Using cached async_timeout-3.0.1-py3-none-any.whl (8.2 kB)
Collecting typing-extensions>=3.6.5
  Using cached typing_extensions-3.7.4.3-py3-none-any.whl (22 kB)
Collecting multidict<7.0,>=4.5
  Downloading multidict-5.0.2-cp37-cp37m-manylinux2014_x86_64.whl (142 kB)
     |████████████████████████████████| 142 kB 2.2 MB/s 
Collecting chardet<4.0,>=2.0
  Using cached chardet-3.0.4-py2.py3-none-any.whl (133 kB)
Collecting idna>=2.0
  Using cached idna-2.10-py2.py3-none-any.whl (58 kB)
Collecting cffi>=1.0.0
  Using cached cffi-1.14.3-cp37-cp37m-manylinux1_x86_64.whl (401 kB)
Processing /home/toby/.cache/pip/wheels/56/b0/fe/4410d17b32f1f0c3cf54cdfb2bc04d7b4b8f4ae377e2229ba0/future-0.18.2-py3-none-any.whl
Collecting six
  Using cached six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting pycparser
  Using cached pycparser-2.20-py2.py3-none-any.whl (112 kB)
Building wheels for collected packages: mautrix-signal, python-olm
  Building wheel for mautrix-signal (setup.py) ... done
  Created wheel for mautrix-signal: filename=mautrix_signal-0.1.0b2+dev.unknown-py3-none-any.whl size=70004 sha256=68dfee5c1ba35c2f153636712edba831d2cba4dcf1e79b5410159caff6d7cd87
  Stored in directory: /tmp/pip-ephem-wheel-cache-n97ur16y/wheels/0b/20/46/8ed37f67ae01472daaddea2d21820d9d81ebad9c7bf6d87a43
  Building wheel for python-olm (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/toby/mautrix-signal-virtualenv-2/bin/python3.7 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-8ekd881k/python-olm/setup.py'"'"'; __file__='"'"'/tmp/pip-install-8ekd881k/python-olm/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-hd2x9s90
       cwd: /tmp/pip-install-8ekd881k/python-olm/
  Complete output (29 lines):
  make: *** No rule to make target '../include/olm/olm.h', needed by 'include/olm/olm.h'.  Stop.
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.7
  creating build/lib.linux-x86_64-3.7/olm
  copying olm/session.py -> build/lib.linux-x86_64-3.7/olm
  copying olm/_finalize.py -> build/lib.linux-x86_64-3.7/olm
  copying olm/__version__.py -> build/lib.linux-x86_64-3.7/olm
  copying olm/pk.py -> build/lib.linux-x86_64-3.7/olm
  copying olm/utility.py -> build/lib.linux-x86_64-3.7/olm
  copying olm/sas.py -> build/lib.linux-x86_64-3.7/olm
  copying olm/account.py -> build/lib.linux-x86_64-3.7/olm
  copying olm/__init__.py -> build/lib.linux-x86_64-3.7/olm
  copying olm/group_session.py -> build/lib.linux-x86_64-3.7/olm
  copying olm/_compat.py -> build/lib.linux-x86_64-3.7/olm
  running build_ext
  generating cffi module 'build/temp.linux-x86_64-3.7/_libolm.c'
  creating build/temp.linux-x86_64-3.7
  building '_libolm' extension
  creating build/temp.linux-x86_64-3.7/build
  creating build/temp.linux-x86_64-3.7/build/temp.linux-x86_64-3.7
  x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.7m -I/home/toby/mautrix-signal-virtualenv-2/include/python3.7m -c build/temp.linux-x86_64-3.7/_libolm.c -o build/temp.linux-x86_64-3.7/build/temp.linux-x86_64-3.7/_libolm.o -I../include
  build/temp.linux-x86_64-3.7/_libolm.c:573:18: fatal error: olm/pk.h: No such file or directory
           #include <olm/pk.h>
                    ^~~~~~~~~~
  compilation terminated.
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for python-olm
  Running setup.py clean for python-olm
Successfully built mautrix-signal
Failed to build python-olm
Installing collected packages: ruamel.yaml.clib, ruamel.yaml, python-magic, commonmark, typing-extensions, multidict, idna, yarl, async-timeout, attrs, chardet, aiohttp, mautrix, asyncpg, pycparser, cffi, future, python-olm, phonenumbers, Pillow, pycryptodome, prometheus-client, six, qrcode, unpaddedbase64, mautrix-signal
    Running setup.py install for python-olm ... error
    ERROR: Command errored out with exit status 1:
     command: /home/toby/mautrix-signal-virtualenv-2/bin/python3.7 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-8ekd881k/python-olm/setup.py'"'"'; __file__='"'"'/tmp/pip-install-8ekd881k/python-olm/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-rcyp88ny/install-record.txt --single-version-externally-managed --compile --install-headers /home/toby/mautrix-signal-virtualenv-2/include/site/python3.7/python-olm
         cwd: /tmp/pip-install-8ekd881k/python-olm/
    Complete output (29 lines):
    make: *** No rule to make target '../include/olm/olm.h', needed by 'include/olm/olm.h'.  Stop.
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.7
    creating build/lib.linux-x86_64-3.7/olm
    copying olm/session.py -> build/lib.linux-x86_64-3.7/olm
    copying olm/_finalize.py -> build/lib.linux-x86_64-3.7/olm
    copying olm/__version__.py -> build/lib.linux-x86_64-3.7/olm
    copying olm/pk.py -> build/lib.linux-x86_64-3.7/olm
    copying olm/utility.py -> build/lib.linux-x86_64-3.7/olm
    copying olm/sas.py -> build/lib.linux-x86_64-3.7/olm
    copying olm/account.py -> build/lib.linux-x86_64-3.7/olm
    copying olm/__init__.py -> build/lib.linux-x86_64-3.7/olm
    copying olm/group_session.py -> build/lib.linux-x86_64-3.7/olm
    copying olm/_compat.py -> build/lib.linux-x86_64-3.7/olm
    running build_ext
    generating cffi module 'build/temp.linux-x86_64-3.7/_libolm.c'
    creating build/temp.linux-x86_64-3.7
    building '_libolm' extension
    creating build/temp.linux-x86_64-3.7/build
    creating build/temp.linux-x86_64-3.7/build/temp.linux-x86_64-3.7
    x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.7m -I/home/toby/mautrix-signal-virtualenv-2/include/python3.7m -c build/temp.linux-x86_64-3.7/_libolm.c -o build/temp.linux-x86_64-3.7/build/temp.linux-x86_64-3.7/_libolm.o -I../include
    build/temp.linux-x86_64-3.7/_libolm.c:573:18: fatal error: olm/pk.h: No such file or directory
             #include <olm/pk.h>
                      ^~~~~~~~~~
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/toby/mautrix-signal-virtualenv-2/bin/python3.7 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-8ekd881k/python-olm/setup.py'"'"'; __file__='"'"'/tmp/pip-install-8ekd881k/python-olm/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-rcyp88ny/install-record.txt --single-version-externally-managed --compile --install-headers /home/toby/mautrix-signal-virtualenv-2/include/site/python3.7/python-olm Check the logs for full command output.
```

Any guesses?

Using Ubuntu 18.04.5, python 3.7.5, python3.7-dev 3.7.5-2~18.04

--- a/Dockerfile
+++ b/Dockerfile
@@ -19,9 +19,10 @@ RUN apk add --no-cache \
       ca-certificates \
       su-exec \
       # encryption
+      libressl \
       olm-dev \
       py3-cffi \
-	  py3-pycryptodome \
+      py3-pycryptodome \
       py3-unpaddedbase64 \
       py3-future \
       bash \
@@ -33,7 +34,7 @@ RUN apk add --no-cache \
 COPY requirements.txt /opt/mautrix-signal/requirements.txt
 COPY optional-requirements.txt /opt/mautrix-signal/optional-requirements.txt
 WORKDIR /opt/mautrix-signal
-RUN apk add --virtual .build-deps python3-dev libffi-dev build-base \
+RUN apk add --virtual .build-deps python3-dev libffi-dev libressl-dev build-base \
  && pip3 install -r requirements.txt -r optional-requirements.txt \
  && apk del .build-deps
 

--- a/mautrix_signal/portal.py
+++ b/mautrix_signal/portal.py
@@ -24,6 +24,8 @@ import os.path
 import time
 import os
 
+from signalstickers_client import StickersClient
+
 from mausignald.types import (Address, MessageData, Reaction, Quote, Group, Contact, Profile,
                               Attachment, GroupID, GroupV2ID, GroupV2)
 from mautrix.appservice import AppService, IntentAPI
@@ -313,9 +315,23 @@ class Portal(DBPortal, BasePortal):
 
         if message.sticker:
             if not message.sticker.attachment.incoming_filename:
-                self.log.warning("Failed to bridge sticker, no incoming filename: %s",
-                                 message.sticker.attachment)
-            else:
+                self.log.debug("Downloading sticker from signal, as no incoming filename was defined: %s",
+                               message.sticker.attachment)
+                try:
+                    async with StickersClient() as client:
+                        sticker_data = await client.download_sticker(message.sticker.sticker_id,
+                                                                     message.sticker.pack_id,
+                                                                     message.sticker.pack_key)
+
+                    path = os.path.join(self.config["signal.outgoing_attachment_dir"],
+                                        f"{message.sticker.pack_id}_{message.sticker.sticker_id}")
+                    with open(path, "wb") as file:
+                        file.write(sticker_data)
+                    message.sticker.attachment.incoming_filename = path
+                except Exception as ex:
+                    self.log.warning("Failed to download sticker: %s", ex)
+
+            if message.sticker.attachment.incoming_filename:
                 content = await self._handle_signal_attachment(intent, message.sticker.attachment)
                 if reply_to:
                     content.set_reply(reply_to)

--- a/requirements.txt
+++ b/requirements.txt
@@ -6,3 +6,4 @@ yarl>=1,<2
 attrs>=19.1
 mautrix>=0.8,<0.9
 asyncpg>=0.20,<0.22
+signalstickers-client>=3.0

